### ISSUE TITLE
YQuake 2 doesn't work anymore on Windows XP

### ISSUE THREAD

**Issue by carlo-bramini** [yquake2/yquake2#994]
I tested the latest released version and it doesn't work anymore on Windows XP.
The problem happens because `wcscat_s()` is not supported on this Operating System.
The fix could be one of these:

1)
Here:
https://github.com/yquake2/yquake2/blob/master/src/backends/windows/system.c#L580
the length into `wpath` is known because it is returned by `MultiByteToWideChar()`, so it could be possible to do something like this:

```
iPathLen=MultiByteToWideChar(CP_UTF8, 0, path, -1, wpath, MAX_OSPATH);
wcsncpy(wpathwithwildcard, MAX_OSPATH, wpath);
wcsncat(wpathwithwildcard, MAX_OSPATH-iPathLen, L"\\*.*");
```

2)
Wine provides an easy and tiny implementation of `wcscat_s()` here:
https://github.com/wine-mirror/wine/blob/master/dlls/msvcrt/wcs.c#L2521
You may just import that function into the source code and rename it to something which won't conflict with the declaration of `wcscat_s()` into the SDK.

After fixing the issue raised by `wcscat_s()`, I recompiled the sources and YQuake 2 worked perfectly again on Windows XP.

**Comment by Yamagi** [comment]
I have implemented your suggestion. A new test build can be found here: https://deponie.yamagi.org/quake2/windows/testbuilds/quake2-20230422-df845e0.zip

**Comment by carlo-bramini** [comment]
There are also other two calls to `wcscat_s()` into the same function:

https://github.com/yquake2/yquake2/blob/df845e0f67bd1b046997555400390c45e067fd32/src/backends/windows/system.c#L598

**Comment by Yamagi** [comment]
Sorry, I overlooked them. Fixed in 22617d881206476184480ea520c319b3676d4208.
